app/controllers/video_controller.rb

```diff
@@ -1,8 +1,19 @@
 class VideoController < ApplicationController
+  #before_action :logged_in_access_only, except: :landing
+  #before_action :logged_out_access_only, only: :landing
+  before_action { |c| c.available_to 'logged_in'}
+
   def home
     @categories = Category.all
   end
   def show
     @video = Video.find(params[:id])
   end
+
+  def search
+    @videos = Video.search_by_title(params[:search])
+    if @videos == []
```

Comment:
Another way to do this is to do `if @videos.none?`. Are you sure you want to redirect just because there are no matches? Usually you just say no matches in the rendered template.